src/SprykerEco/Zed/AmazonQuicksight/Business/Matcher/QuicksightUserMatcher.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AmazonQuicksight\Business\Matcher;
+
+use ArrayObject;
+use SprykerEco\Zed\AmazonQuicksight\Business\Filter\QuicksightUserCollectionFilterInterface;
+use SprykerEco\Zed\AmazonQuicksight\Business\Filter\UserCollectionFilterInterface;
+use SprykerEco\Zed\AmazonQuicksight\Business\Reader\UserReaderInterface;
+
+class QuicksightUserMatcher implements QuicksightUserMatcherInterface
```

Comment:
The behaviour if this class looks like a filter? What do you think, maybe it is better to get rid of it and move to filter?

Reply:
I was thinking about it, but the problem is I'm adding `fkUser` to `QuicksightUserTransfer` here, which is not really filter responsibility, that's why I decided to introduce a separate class.